recipe/meta.yaml

```diff
@@ -1,41 +1,61 @@
 {% set name = "c-blosc2" %}
 {% set version = "2.14.4" %}
 
+# update this when ABI version changes
+{% set abi_version = "3" %}
+
 package:
-  name: {{ name }}
+  name: {{ name }}-pkg
   version: {{ version }}
 
 source:
   url: https://github.com/Blosc/{{ name }}/archive/v{{ version }}.tar.gz
   sha256: b5533c79aacc9ac152c80760ed1295a6608938780c3e1eecd7e53ea72ad986b0
 
 build:
-  number: 0
-  run_exports:
-    - {{ pin_subpackage('c-blosc2') }}
-
-requirements:
-  build:
-    - cmake
-    - make  # [unix]
-    - {{ compiler('c') }}
-    - {{ compiler('cxx') }}
-  host:
-    - lz4-c
-    - zstd
-    - zlib-ng
-
-test:
-  commands:
-    - test -e $PREFIX/include/blosc2.h                    # [unix]
-    - test -e $PREFIX/include/blosc2/blosc2-export.h      # [unix]
-    - test -e $PREFIX/lib/libblosc2${SHLIB_EXT}           # [unix]
-    - test -e $PREFIX/lib/pkgconfig/blosc2.pc             # [unix]
-    - if not exist %LIBRARY_INC%\blosc2.h exit 1                # [win]
-    - if not exist %LIBRARY_INC%\blosc2\blosc2-export.h exit 1  # [win]
-    - if not exist %LIBRARY_BIN%\libblosc2.dll exit 1           # [win]
-    - if not exist %LIBRARY_LIB%\pkgconfig\blosc2.pc exit 1     # [win]
-    - if not exist %LIBRARY_LIB%\blosc2.lib exit 1.             # [win]
+  number: 1
+
+outputs:
+  - name: c-blosc2
+    build:
+      script: ${RECIPE_DIR}/build-blosc2.sh  # [not win]
+      script: "%RECIPE_DIR%\\bld-blosc2.bat"  # [win]
+      run_exports:
+        - {{ pin_subpackage('c-blosc2') }}
+        - {{ pin_subpackage('c-blosc2-abi') }}
+  
+    requirements:
+      build:
+        - cmake
+        - make  # [unix]
+        - {{ compiler('c') }}
+        - {{ compiler('cxx') }}
+      host:
+        - lz4-c
+        - zstd
+        - zlib-ng
+      run:
+        - {{ pin_subpackage('c-blosc2-abi', exact=True) }}
+
+    test:
+      commands:
+        - test -e $PREFIX/include/blosc2.h                    # [unix]
+        - test -e $PREFIX/include/blosc2/blosc2-export.h      # [unix]
+        - test -e $PREFIX/lib/libblosc2${SHLIB_EXT}           # [unix]
+        - test -e $PREFIX/lib/libblosc2.{{ abi_version }}${SHLIB_EXT}  # [osx]
+        - test -e $PREFIX/lib/libblosc2${SHLIB_EXT}.{{ abi_version }}  # [linux]
+        - test -e $PREFIX/lib/pkgconfig/blosc2.pc             # [unix]
+        - if not exist %LIBRARY_INC%\blosc2.h exit 1                # [win]
+        - if not exist %LIBRARY_INC%\blosc2\blosc2-export.h exit 1  # [win]
+        - if not exist %LIBRARY_BIN%\libblosc2.dll exit 1           # [win]
+        - if not exist %LIBRARY_LIB%\pkgconfig\blosc2.pc exit 1     # [win]
+        - if not exist %LIBRARY_LIB%\blosc2.lib exit 1.             # [win]
+  
+  # ABI package to pinning,
+  # since there's no relationship between c-blosc package version and ABI version
+  - name: c-blosc2-abi
```

Comment:
Do you want to add yourself as a maintainer?

Personally, I don't have the bandwidth to maintain these split ABI package and without extra support I would prefer a "keeping it simple" solution: closer to https://github.com/conda-forge/c-blosc2-feedstock/pull/63